src/toxcore/dht/packet/ping_request.rs

```diff
@@ -48,14 +66,15 @@ impl ToBytes for PingRequest {
     }
 }
 
-impl FromBytes for PingRequest {
-    named!(from_bytes<PingRequest>, do_parse!(
-        tag!("\x00") >>
-        pk: call!(PublicKey::from_bytes) >>
-        nonce: call!(Nonce::from_bytes) >>
-        payload: map!(rest, |bytes| bytes.to_vec() ) >>
-        (PingRequest { pk, nonce, payload })
-    ));
+impl FromBytes5 for PingRequest {
+    fn from_bytes5<'a, E: ParseError<&'a [u8]>>(i: &'a [u8]) ->IResult<&'a [u8], PingRequest, E> {
+        map(tuple((
```

Comment:
Namsoo, in the best case we should migrate to nom5 only by changing the version of `nom` in out Cargo.toml with almost no changes.

So the code here should be as close to `FromBytes` as possible. With macro named and do_parse.

Reply:
If these macros are missing in nom5 you have to find a way to implement them